### PR TITLE
Update plotly.js to 2.24.1 and add docs for new features

### PR DESCRIPTION
**Pull request by LiamConnors**
### Documentation PR

- [x] I've [seen the `doc/README.md` file](https://github.com/plotly/plotly.py/blob/master/doc/README.md)
- [x] This change runs in the current version of Plotly on PyPI and targets the `doc-prod` branch OR it targets the `master` branch
- [x] If this PR modifies the first example in a page or adds a new one, it is a `px` example if at all possible
- [x] Every new/modified example has a descriptive title and motivating sentence or paragraph
- [x] Every new/modified example is independently runnable
- [x] Every new/modified example is optimized for short line count	and focuses on the Plotly/visualization-related aspects of the example rather than the computation required to produce the data being visualized
- [x] Meaningful/relatable datasets are used for all new examples instead of randomly-generated data where possible
- [x] The random seed is set if using randomly-generated data in new/modified examples
- [x] New/modified remote datasets are loaded from https://plotly.github.io/datasets and added to https://github.com/plotly/datasets
- [x] Large computations are avoided in the new/modified examples in favour of loading remote datasets that represent the output of such computations
- [x] Imports are `plotly.graph_objects as go` / `plotly.express as px` / `plotly.io as pio`
- [x] Data frames are always called `df`
- [x] `fig = <something>` call is high up in each new/modified example (either `px.<something>` or `make_subplots` or `go.Figure`)
- [x] Liberal use is made of `fig.add_*` and `fig.update_*` rather than `go.Figure(data=..., layout=...)` in every new/modified example
- [x] Specific adders and updaters like `fig.add_shape` and `fig.update_xaxes` are used instead of big `fig.update_layout` calls in every new/modified example
- [x] `fig.show()` is at the end of each new/modified example
- [x] `plotly.plot()` and `plotly.iplot()` are not used in any new/modified example
- [x] Hex codes for colors are not used in any new/modified example in favour of [these nice ones](https://github.com/plotly/plotly.py/issues/2192)

## Code PR

- [x] I have read through the [contributing notes](https://github.com/plotly/plotly.py/blob/master/contributing.md) and understand the structure of the package. In particular, if my PR modifies code of `plotly.graph_objects`, my modifications concern the `codegen` files and not generated files.
- [ ] I have added tests (if submitting a new feature or correcting a bug) or
  modified existing tests.
- [x] For a new feature, I have added documentation examples in an existing or
  new tutorial notebook (please see the doc checklist as well).
- [x] I have added a CHANGELOG entry if fixing/changing/adding anything substantial.
- [ ] For a new feature or a change in behaviour, I have updated the relevant docstrings in the code to describe the feature or behaviour (please see the doc checklist as well).

-->

### CODE REVIEW

**Review by alexcjohnson — APPROVED** [review]
💃

**Comment by LiamConnors** [comment]
> Looks like the `release_build` job is failing due to jupyterlab 4 updates?

Yeah, looks like something to do with that.

**Comment by alexcjohnson** [comment]
It can wait until @LiamConnors is back, this problem has been around a long time.

**Comment by LiamConnors** [comment]
@alexcjohnson and @archmoj this is ready for another look. I've updated to Plotly.js 2.24.0 and added examples for patterns for the new trace types. Lots of formatting changes applied by the code generator. Not sure why.

File: doc/python/pie-charts.md [hunk 1]
```diff
@@ -300,6 +300,32 @@ fig.update_layout(title_text='World GDP')
 fig.show()
 ```
 
+### Pattern Fills
+
+*New in 5.15*
+
+Pie charts support [patterns](/python/pattern-hatching-texture/) (also known as hatching or texture) in addition to color.
```
[the remark this answers is not in the document]
Comment (LiamConnors) [reply doc/python/pie-charts.md:307]: Added

**Comment by LiamConnors** [comment]
> 🥇 Looking very good to me. Thank you 🙏 @LiamConnors

Thanks @archmoj. 
Some strange behaviour happening with the docs build. Seems unrelated to this PR. Investigating.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,20 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [UNRELEASED]
 
+
+### Updated
+  - Updated Plotly.js from version 2.20.0 to version 2.24.0. See the [plotly.js CHANGELOG](https://github.com/plotly/plotly.js/blob/master/CHANGELOG.md#2240----2023-06-06) for more information. Notable changes include:
```
Comment (alexcjohnson) [inline CHANGELOG.md:9]:
```suggestion
  - Updated Plotly.js from version 2.20.0 to version 2.24.1. See the [plotly.js CHANGELOG](https://github.com/plotly/plotly.js/blob/master/CHANGELOG.md#2241----2023-06-07) for more information. Notable changes include:
```

**Review by alexcjohnson — APPROVED** [review]
💃 after updating the changelog

File: CHANGELOG.md (hunk 2 above)
Comment (LiamConnors) [reply CHANGELOG.md:9]: Ah. Good catch. Thanks.

**Pull request merged**